apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,435 @@
+import { FormValue } from "@wso2is/forms";
+import { Button, GenericIcon, Hint, PrimaryButton, SecondaryButton } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import { AxiosError } from "axios";
+import isEmpty from "lodash-es/isEmpty";
+import React, { ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { AnyAction } from "redux";
+import { ThunkDispatch } from "redux-thunk";
+import { Form, Grid, Icon, List } from "semantic-ui-react";
+import { updateProfileInfo } from "../../../../api";
+import { getAccountRecoveryIcons } from "../../../../configs";
+import { CommonConstants } from "../../../../constants";
+import { commonConfig } from "../../../../extensions";
+import { AlertInterface, AlertLevels, BasicProfileInterface, MultiValue, ProfileSchema } from "../../../../models";
+import { AppState } from "../../../../store";
+import { getProfileInformation, setActiveForm } from "../../../../store/actions";
+import { EditSection } from "../../../shared";
+import "./sms-recovery.scss";
+
+/**
+ * SMS key.
+ */
+const SMS: string = "sms";
+
+/**
+ * Prop types for the SMSRecoveryComponent component.
+ * Also see {@link SMSRecovery.defaultProps}
+ */
+interface SMSRecoveryProps extends IdentifiableComponentInterface {
+    onAlertFired: (alert: AlertInterface) => void;
+    enableEditMobile?: boolean;
+}
+
+/**
+ * SMS recovery section.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const SMSRecovery: React.FunctionComponent<SMSRecoveryProps> = (
+    props: SMSRecoveryProps
+): ReactElement => {
+
+    const {
+        onAlertFired,
+        enableEditMobile,
+        ["data-componentid"]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
+
+    const profileInfo: BasicProfileInterface = useSelector(
+        (state: AppState) => state.authenticationInformation.profileInfo
+    );
+    const phoneNumberSchema: ProfileSchema = useSelector((state: AppState) => {
+        const phoneNumberSchemas: ProfileSchema = state.authenticationInformation.profileSchemas.find(
+            (profileSchema: ProfileSchema) => {
+                return profileSchema.name === "phoneNumbers";
+            });
+
+        if (phoneNumberSchemas && phoneNumberSchemas.subAttributes) {
+            const mobileSchema: ProfileSchema = phoneNumberSchemas.subAttributes?.find(
+                (subAttribute: ProfileSchema) => {
+                    return subAttribute.name === "mobile";
+                });
+
+            if (mobileSchema) return mobileSchema;
+        }
+
+        return phoneNumberSchemas;
+    });
+    const activeForm: string = useSelector((state: AppState) => state.global.activeForm);
+
+    const [ mobile, setMobile ] = useState("");
```

Comment:
use a type for the state

Reply:
use undefined instead of empty string

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/6418/commits/0dd77c8c16b11b96a91e6b1331294e3bb8560a5b